### PR TITLE
Rlc 3: Write led_rgb module with tdd

### PR DESCRIPTION
**Pull request by dtrochow**

### CODE REVIEW

File: new_led_cube/include/led_rgb.hpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#include <iostream>
```
Comment (kjeznach) [inline new_led_cube/include/led_rgb.hpp:1]: Is it really needed? I don't see we use instructions from this library

File: unit/3d_matrix_test.cpp [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#include <gtest/gtest.h>
+
+#include <iostream>
```
Comment (kjeznach) [inline unit/3d_matrix_test.cpp]: Is it really needed?

File: unit/3d_matrix_test.cpp [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+using namespace std;
```
Comment (kjeznach) [inline unit/3d_matrix_test.cpp]: Is it really needed?

File: new_led_cube/include/led_rgb.hpp (hunk 1 above)
Comment (kjeznach) [inline new_led_cube/include/led_rgb.hpp]: Is it really needed? I can't see we're using instructions from this library

File: unit/animations_test.cpp [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#include <gtest/gtest.h>
+
+#include <iostream>
```
Comment (kjeznach) [inline unit/animations_test.cpp]: Is it really needed? I can't see we're using instructions from this library

File: unit/animations_test.cpp [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+
+using namespace std;
```
Comment (kjeznach) [inline unit/animations_test.cpp]: Is it really needed? I can't see we're using instructions from this library

File: unit/led_rgb_test.cpp [hunk 6]
```diff
@@ -0,0 +1,19 @@
+#include <gtest/gtest.h>
+
+#include <iostream>
```
Comment (kjeznach) [inline unit/led_rgb_test.cpp]: Is it really needed? I can't see we're using instructions from this library

File: unit/led_rgb_test.cpp [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+#include "led_rgb.hpp"
+
+using namespace std;
```
Comment (kjeznach) [inline unit/led_rgb_test.cpp]: Is it really needed? I can't see we're using instructions from this library

**Review by kjeznach — COMMENTED** [review]
Hi, there are some things here that I'm not pretty sure about. Can you check this out?

File: CMakeLists.txt [hunk 8]
```diff
@@ -13,14 +13,6 @@ pico_sdk_init()
 add_subdirectory(lib)
 add_subdirectory(led_cube)
 
-# Directories for mcp23017 library
```
Comment (dtrochow) [inline CMakeLists.txt:16]: COMMENT: They are no longer needed.

File: build_run_utests.sh [hunk 9]
```diff
@@ -1,5 +1,13 @@
 #!/bin/bash
-cd unit
-cmake -S . -B build
-cmake --build build
-cd build && ctest
+function build {
```
Comment (dtrochow) [inline build_run_utests.sh:2]: COMMENT: Modifying build and run script to fail if build step fail.

File: new_led_cube/include/colors_config.hpp [hunk 10]
```diff
@@ -0,0 +1,18 @@
+#include <map>
```
Comment (dtrochow) [inline new_led_cube/include/colors_config.hpp:1]: COMMENT: File with configuration of all available led colors.

**Review by kjeznach — APPROVED** [review]
Functional review done.

**Pull request merged**